AK/StringUtils.cpp

```diff
@@ -401,6 +401,18 @@ Vector<size_t> find_all(StringView haystack, StringView needle)
     return positions;
 }
 
+Vector<size_t> find_all(StringView haystack, Function<bool(char)> const& predicate)
+{
+    Vector<size_t> positions;
+    size_t current_position = 0;
+    while (current_position <= haystack.length()) {
```

Comment:
Shouldn't this be `current_position < haystack.length()`? Otherwise it'll index one passed the end of the string (and probably hit a `VERIFY()`

Reply:
Huh, it seems StringView does not check the index is valid...

Reply:
That's right, I coded too fast ;(

Reply:
This highlights the need for unit tests, please add some to `Tests/AK/TestStringView.cpp`